CMakeLists.txt

```diff
@@ -341,8 +341,16 @@ endif()
 ###############################################################################
 # Define compilation and link flags
 
+option(OCIO_BUILD_POSITION_INDEPENDENT_EXECUTABLES "Set to ON to build executables as PIE" OFF)
+
 include(CompilerFlags)
 
+if(OCIO_BUILD_POSITION_INDEPENDENT_EXECUTABLES AND NOT CMAKE_C_LINK_PIE_SUPPORTED)
+    message(FATAL_ERROR "PIE is not supported at link time.\n")
+else()
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+endif()
```

Comment:
Is this logic what you intend? I think this means that if PIE linkage is supported, you are going to turn on CMAKE_POSITION_INDEPENDENT_CODE unconditionally, regardless of how OCIO_BUILD_POSITION_INDEPENDENT_EXECUTABLES was set.

Reply:
yes, looks like I committed a bit of a noob logic error. Serves me right for not testing on a machine that would fail the 2nd sub expression